### PR TITLE
Where buy frontend

### PR DESCRIPTION
**Pull request by setar92**
Please, make review after PR 223, 
https://github.com/BinaryStudioAcademy/bsa-2022-autoline/pull/223

https://gyazo.com/9e7600f3ad967e7158a77580a551263b
https://gyazo.com/e3b13d3a4aa500338aee64a59b397d52
https://gyazo.com/164f02d4cc88c889698de99257ae0615

### CODE REVIEW

File: backend/src/common/types/where-to-buy/where-to-buy.params.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      count: number;
+      last_id: number;
+    };
+  };
+}
+
+type WhereBuyRequestQuery = {
+  page: string;
+  id: string;
+  countpage: string;
```
Comment (RomanMahola) [inline backend/src/common/types/where-to-buy/where-to-buy.params.ts:15]: Do we need smth like 
page_from
page_to 
?

File: backend/src/helpers/cars/api-autoria.helper.ts [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
-  const response = await axios.get(URL, {
+): Promise<AutoriaResponse | undefined> => {
+  const response = await axios.get(AutoRiaUrls.SEARCH_URL, {
+    params: {
+      api_key: ENV.APP.AUTORIA_API_KEY,
+      ...requestParams,
+    },
+  });
+  return response.data;
+};
```
Comment (RomanMahola) [inline backend/src/helpers/cars/api-autoria.helper.ts:17]: I think we also need to handle exception here

File: backend/src/helpers/cars/api-autoria.helper.ts [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
+
+const getCarByIdRia = async (id: string): Promise<Response | undefined> => {
+  const requestParams = { auto_id: id };
+  const response = await axios.get(AutoRiaUrls.INFO_URL, {
```
Comment (RomanMahola) [inline backend/src/helpers/cars/api-autoria.helper.ts:21]: Also here

File: backend/src/services/cars/cars-search.service.ts [hunk 4]
```diff
@@ -3,13 +3,14 @@ import { prisma } from '@data/prisma-client';
 
 const carsSearchAutoria = async (
   complectationId: string,
-  page = '1',
+  page = '0',
```
Comment (RomanMahola) [inline backend/src/services/cars/cars-search.service.ts:6]: why string?

File: backend/src/services/cars/cars-search.service.ts [hunk 5]
```diff
@@ -3,13 +3,14 @@ import { prisma } from '@data/prisma-client';
...
+  countpage = 10,
 ): Promise<Partial<AutoriaRequestParams>> => {
   const result: Partial<AutoriaRequestParams> = {
     category_id: 1,
     power_name: 1,
```
Comment (RomanMahola) [inline backend/src/services/cars/cars-search.service.ts:11]: Is that correct?

File: backend/src/services/where-buy/cars-properties.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  fuel_type: {
+    autoria_code: number;
+  };
+  transmission_type: {
+    autoria_code: number;
+  };
+}
+
+const carProperties = async (id: string): Promise<PropertiesProp | null> => {
+  console.log('properties');
```
Comment (RomanMahola) [inline backend/src/services/where-buy/cars-properties.ts]: Delete console.log()

File: frontend/src/components/where-to-buy/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+.noFoundContainer {
+  display: flex;
+  margin: 30px;
+  padding: 25px;
+  background-color: var(--white);
+  border: solid 1px var(--white);
+  border-radius: 5px;
+
+  h3 {
```
Comment (RomanMahola) [inline frontend/src/components/where-to-buy/styles.module.scss]: Try to always use class

File: frontend/src/components/where-to-buy/styles.module.scss [hunk 8]
```diff
@@ -0,0 +1,70 @@
+.container {
+  max-width: 1250px;
```
Comment (RomanMahola) [inline frontend/src/components/where-to-buy/styles.module.scss]: This is bad for high resolution

File: frontend/src/components/where-to-buy/where-buy-item/styles.module.scss [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+        padding: 0 10px;
+        overflow: hidden;
+      }
+
+      .textDescription.more {
+        display: inline-block;
+      }
+
+      div::before {
+        content: "\2022";
```
Comment (RomanMahola) [inline frontend/src/components/where-to-buy/where-buy-item/styles.module.scss:104]: We would always have 2022? 
Still in 2025?

File: frontend/src/components/where-to-buy/where-buy-item/where-buy-item.tsx [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+      <div className={styles.container}>
+        <div className={styles.logo}>
+          <img src={logo} alt="logo" />
+        </div>
+        <div className={styles.description}>
+          <span className={styles.carName}>{title}</span>
+          <div className={styles.labelGroup}>
+            <div className={styles.label}>
+              <img src={speedometr} alt="speedometr" />
+              {`${race} 000 km`}
```
Comment (RomanMahola) [inline frontend/src/components/where-to-buy/where-buy-item/where-buy-item.tsx:57]: Are you sure that race always would be in km not miles?

File: frontend/src/components/where-to-buy/where-buy-item/where-buy-item.tsx [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+            <div className={styles.trigger}>
+              <div
+                ref={elementRef}
+                className={clsx(styles.textDescription, {
+                  [(styles.textDescription, styles.more)]: open,
+                })}
+              >
+                {description}
+              </div>
+              {height > 47 && (
```
Comment (RomanMahola) [inline frontend/src/components/where-to-buy/where-buy-item/where-buy-item.tsx]: What is 47?

Comment (RomanMahola) [reply frontend/src/components/where-to-buy/where-buy-item/where-buy-item.tsx]: Use const

File: frontend/src/components/where-to-buy/where-to-buy.tsx [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+  complectationId: string;
+}
+
+const WhereToBuy: React.FC<WhereToBuyProps> = ({ complectationId }) => {
+  const [page, setPage] = useState(0);
+  const [isSorted, setIsSorted] = useState(false);
+  useGetWhereBuyQuery({
+    page,
+    complectationId,
+    countpage: '20',
```
Comment (RomanMahola) [inline frontend/src/components/where-to-buy/where-to-buy.tsx]: Use variable here. Maybe in future we would provide section how many items on the page user want to see

File: frontend/src/components/where-to-buy/where-to-buy.tsx [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+    );
+  }
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.title}>
+        <div className={styles.carName}>
+          {advertsList &&
+            advertsList[0] &&
+            `${advertsList[0].markName} ${advertsList[0].modelName}`}
```
Comment (RomanMahola) [inline frontend/src/components/where-to-buy/where-to-buy.tsx]: Simplify it

File: frontend/src/store/queries/where-buy.ts [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+          id: complectationId,
+          countpage,
+        },
+      }),
+      async onQueryStarted({ complectationId }, { dispatch, queryFulfilled }) {
+        try {
+          const { data: adverts } = await queryFulfilled;
+          dispatch(setAdverts({ adverts, complectationId }));
+        } catch (err) {
+          console.error(err);
```
Comment (RomanMahola) [inline frontend/src/store/queries/where-buy.ts]: Please don't use `console.log()` Users won't see problems in console. 
We also have linter rule about

File: frontend/src/store/where-buy/slice.ts [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+  complectationId: '',
+};
+
+const { reducer, actions } = createSlice({
+  name: 'advert',
+  initialState,
+  reducers: {
+    setAdverts: (state, action) => {
+      const { adverts, complectationId } = action.payload;
+      if (state.complectationId == complectationId) {
```
Comment (RomanMahola) [inline frontend/src/store/where-buy/slice.ts]: Use === instead of ==

**Review by RomanMahola — CHANGES_REQUESTED** [review]

File: backend/src/helpers/cars/api-autoria.helper.ts (hunk 3 above)
Comment (setar92) [reply backend/src/helpers/cars/api-autoria.helper.ts:21]: we will catch this error above

File: frontend/src/components/where-to-buy/where-buy-item/styles.module.scss (hunk 9 above)
Comment (setar92) [reply frontend/src/components/where-to-buy/where-buy-item/styles.module.scss:104]: it is not a year, it is code of
https://gyazo.com/985212dd79efca44004f2e0e3c0b4975

File: frontend/src/components/where-to-buy/where-buy-item/where-buy-item.tsx (hunk 10 above)
Comment (setar92) [reply frontend/src/components/where-to-buy/where-buy-item/where-buy-item.tsx:57]: Yes, it is according autoria documantation.

File: backend/src/services/cars/cars-search.service.ts (hunk 5 above)
Comment (setar92) [reply backend/src/services/cars/cars-search.service.ts:11]: Yes, it is correct:
 category_id: 1 mean "cars"
 power_name: 1 mean "Power unit - hp"

**Pull request merged**